plutus-playground/plutus-playground-server/src/Playground/Interpreter.hs

```diff
@@ -1,166 +1,195 @@
-    , OverloadedStrings
-    , RecordWildCards
-    , ScopedTypeVariables
-    , StandaloneDeriving
-    , TemplateHaskell
-    ]
-
-loadSource :: (MonadInterpreter m) => FilePath -> (ModuleName -> m a) -> m a
-loadSource fileName action =
-    flip finally (liftIO (removeFile fileName)) $ do
-        set [languageExtensions := defaultExtensions]
-        loadModules [fileName]
-        (m:_) <- getLoadedModules
-        setTopLevelModules [m]
-        action m
+import           Control.Monad              (unless)
+import           Control.Monad.Catch        (MonadCatch, MonadMask, bracket, catch)
+import           Control.Monad.Error.Class  (MonadError, throwError)
+import           Control.Monad.IO.Class     (MonadIO, liftIO)
+import qualified Control.Newtype.Generics   as Newtype
+import           Data.Aeson                 (ToJSON)
+import qualified Data.Aeson                 as JSON
+import           Data.ByteString            (ByteString)
+import qualified Data.ByteString.Char8      as BS8
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import           Data.List                  (intercalate)
+import           Data.Swagger               (Schema)
+import           Data.Text                  (Text)
+import qualified Data.Text                  as Text
+import qualified Data.Text.Internal.Search  as Text
+import qualified Data.Text.IO               as Text
+import           Ledger.Types               (Blockchain, Value)
+import           Playground.API             (Evaluation (sourceCode), Expression (Action, Wait), Fn (Fn),
+                                             FunctionSchema,
+                                             PlaygroundError (CompilationErrors, DecodeJsonTypeError, InterpreterError, OtherError),
+                                             SourceCode, parseErrorsText, program, wallets)
+import           System.Directory           (removeFile)
+import           System.Environment         (lookupEnv)
+import           System.Exit                (ExitCode (ExitSuccess))
+import           System.IO                  (Handle, hClose, hFlush)
+import           System.IO.Temp             (getCanonicalTemporaryDirectory, openTempFile)
+import           System.Process             (readProcessWithExitCode)
+import qualified Text.Regex                 as Regex
+import           Wallet.Emulator.Types      (EmulatorEvent, Wallet)
+
+run :: (MonadIO m, MonadError PlaygroundError m) => FilePath -> m String
+run scriptPath = do
+    runghc <- lookupRunghc
+    (exitCode, stdout, stderr) <-
+        liftIO $ readProcessWithExitCode runghc [scriptPath] ""
+    case exitCode of
+        ExitSuccess -> pure stdout
+        _           -> throwError . CompilationErrors $ parseErrorsText (Text.pack stderr)
+
+replaceModuleName :: Text -> Text
+replaceModuleName script =
+    let scriptString = Text.unpack script
+        regex = Regex.mkRegex "module .* where"
+     in Text.pack $ Regex.subRegex regex scriptString "module Main where"
```

Comment:
This will murder export lists, but that's probably fine.

Reply:
Yeah, I see these contracts as a Main module anyway, so export lists should be irrelevant